tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks
+
+import dagr.core.tasksystem.{Pipeline, Task}
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Provides a set of objects and classes for composing Scatter/Gather tasks and pipelines.
+ */
+object ScatterGather {
+
+  /**
+   * A type of Task that, when run, partitions some kind of input, and produces one or more outputs
+   * of type A that should be parallelized (scattered) over.
+   */
+  trait Scatterer[A] extends Task {
```

Comment:
I like the one word name and something short: `Splitter`, `Divider`, `Separator`, `Subdivider`.  We can keep `Scatterer` for now.

As per your input example (`Scatterer[(PathToBam, PathToFasta, PathToIntervals)`), I would say pick what is the primary input data source, and this case likely the `PathToIntervals`.  While not ideal, I don't think we want ``Scatterer[(PathToBam, PathToFasta, PathToIntervals, Int, Int, Int, Int, Float, Float)` where we include all of the parameters to the scatterer.  I think this is more subjective and meant for clarity, rather than a rule.